### PR TITLE
feat(configurablebranches): adds functionality for configurable branches

### PR DESCRIPTION
**Pull request by staciegravy**
An example of how to configure this in dinghy.yml:

```
repoConfig:
  - branch: refs/heads/test_branch
    provider: bitbucket-server
    repo: stacie-repo
```

### CODE REVIEW

**Review by dotdotdotpaul — APPROVED** [review]
Let's see if it works. 😄

File: pkg/git/github/push.go [hunk 1]
```diff
@@ -101,7 +101,17 @@ func (p *Push) Org() string {
...
 }
 
+// Branch returns the branch of the push
+func (p *Push) Branch() string {
+	return p.Ref
+}
+
 // IsMaster detects if the branch is master.
 func (p *Push) IsMaster() bool {
 	return p.Ref == "refs/heads/master"
```
Comment (dotdotdotpaul) [inline pkg/git/github/push.go:111]: Doesn't this need to reflect what "master" is configured to, for the repo?

**Review by dotdotdotpaul — CHANGES_REQUESTED** [review]

File: pkg/git/github/github.go [hunk 2]
```diff
@@ -50,14 +49,15 @@ func newGitHubClient(ctx context.Context, endpoint, token string) (*github.Clien
...
-func (g *GitHub) DownloadContents(org, repo, path string) (string, error) {
+func (g *GitHub) DownloadContents(org, repo, path, branch string) (string, error) {
 	ctx := context.Background()
 	client, err := newGitHubClient(ctx, g.Endpoint, g.Token)
 	if err != nil {
 		return "", err
 	}
 
-	r, err := client.Repositories.DownloadContents(ctx, org, repo, path, nil)
+	opt := github.RepositoryContentGetOptions{Ref: branch}
```
Comment (jeffhart-armory) [inline pkg/git/github/github.go]: Any reason not to use `&github.RepositoryContentGetoptions` and pass in the variable instead of the reference?

**Review by dotdotdotpaul — APPROVED** [review]

File: pkg/git/github/github.go (hunk 2 above)
Comment (staciegravy) [reply pkg/git/github/github.go]: The library that we're using requires that `RepositoryContentGetOptions` type to be passed in. Could prob rename some of the Github stuff in there just to make it less confusing as to what is ours and what is the lib.

**Review by jeffhart-armory — APPROVED** [review]

**Pull request merged**